controlplane/kubeadm/controllers/upgrade.go

```diff
@@ -105,7 +106,10 @@ func (r *KubeadmControlPlaneReconciler) upgradeControlPlane(
 		return ctrl.Result{}, err
 	}
 
-	if status.Nodes <= *kcp.Spec.Replicas {
+	// RolloutStrategy is currently defaulted and validated to be RollingUpdate
+	// We can ignore MaxUnavailable because we are enforcing health checks before we get here.
+	maxNodes := *kcp.Spec.Replicas + int32(kcp.Spec.RolloutStrategy.RollingUpdate.MaxSurge.IntValue())
```

Comment:
@detiber @sedefsavas @CecileRobertMichon @fabriziopandini Please double check these changes, given that you all have worked more or less on this codebase.

If we were to backport this to v1alpha3, would you foresee any issues?